### PR TITLE
Added VB (specific/focused) SourceGenerators Sample

### PR DESCRIPTION
**Pull request by DualBrain**
This is a VB focused example (similar to the original C#).  It's been cleaned up and VB'ised in many areas.

I'm submitting this pull request in part based on the recommendation of @KathleenDollard - if there's anything that needs to be tweaked, adjusted, etc., please let me know.

Thanks for your consideration.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/roslyn-sdk?pullRequest=691) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/roslyn-sdk?pullRequest=691) before we can accept your contribution.<br/><br/>:x: DualBrain [sign now](https://cla.dotnetfoundation.org/dotnet/roslyn-sdk?pullRequest=691)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/roslyn-sdk?pullRequest=691) it.</sub>

**Comment by DualBrain** [comment]
Please note that this isn't the same approach that @sharwell has done (https://github.com/dotnet/roslyn-sdk/pull/682); his approach provides value to those that want to handle multiple languages.  This pull request is from the point of view of the VB dev wanting to create their own source generators in VB for consumption by VB devs; so I see value in having both examples.

File: samples/VisualBasic/SourceGenerators/GeneratedDemo/bin/Debug/netcoreapp3.1/GeneratedDemo.deps.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (sharwell) [inline samples/VisualBasic/SourceGenerators/GeneratedDemo/bin/Debug/netcoreapp3.1/GeneratedDemo.deps.json:1]: ❗ This pull request must be rebased to remove the bin/ and obj/ directories from _all commits_ involved in this pull request (we cannot have them in the repository history).

**Review by sharwell — CHANGES_REQUESTED** [review]

**Comment by sharwell** [comment]
I have some doubts about the incremental value and maintenance burden imposed by these samples over the ones already implemented. At the end of the day, it's exactly the same concept as what is already in the repository, except mechanically ported to Visual Basic. I have doubts about the number of VB users who would be unable to use Source Generators solely due to the lack having access to pre-ported code.

File: samples/VisualBasic/SourceGenerators/GeneratedDemo/bin/Debug/netcoreapp3.1/GeneratedDemo.deps.json (hunk 1 above)
Comment (DualBrain) [reply samples/VisualBasic/SourceGenerators/GeneratedDemo/bin/Debug/netcoreapp3.1/GeneratedDemo.deps.json:1]: Didn't realize there wasn't a git.ignore for this project, so for that I apologize for assuming... I can redo... however... given the other comment... doesn't sound like there is interest in accepting the pull.

**Comment by DualBrain** [comment]
Doesn't bother me either way... I've already made these available at https://github.com/DualBrain/Samples/tree/master/SourceGenerators.  Additionally, it was only, in part, "mechanically ported" - a lot of hands on work went in to march it forward as being more VB-ish.  In any case, I was asked to submit it... if it's not desired, no you won't hear any boohooing from me.

**Comment by DualBrain** [comment]
Will leave another note regarding potential "value"... want to point out that the **existing** samples use many **very C# specific concepts/language features** that are absolutely **ALIEN** to, well, **ALL** VB devs who don't have any interest in learning (and keeping up with the constant) C# changes.  Off the top of my head there is usage of pattern matching, tuple deconstruction, nullable, etc.  None of this is in VB and isn't planned on being there any time soon.  So if the desire is to not have VB specific samples... please reconsider *how* the code is written for examples *showing concepts* so that concepts/features that do not exist in VB aren't utilized all over the place.  At least, that way, a mechanical conversion could be reasonable/feasible. 

CC: @KathleenDollard

**Comment by KathleenDollard** [comment]
It was me that requested @DualBrain submit this sample.

Sam, you've got some good points. I think we should chat about samples after the first of the year. As C# samples embrace new styles, the point about it becoming more and more difficult to convert is real. Also, VB samples will not need to be updated to keep them from _looking_ stale, as the language isn't expected to change. They could become out of date when we obsolete or offer new approaches, of course.

File: samples/VisualBasic/SourceGenerators/GeneratedDemo/bin/Debug/netcoreapp3.1/GeneratedDemo.deps.json (hunk 1 above)
Comment (jmarolf) [reply samples/VisualBasic/SourceGenerators/GeneratedDemo/bin/Debug/netcoreapp3.1/GeneratedDemo.deps.json:1]: feel free to add obj/bin folders to the current .gitignore file.

**Comment by jmarolf** [comment]
We already have duplicate samples for both VB and C# in all other areas. I am fine accepting this pull request in principle.

**Comment by DualBrain** [comment]
Resubmitted as a different pull request since I couldn't figure out how to "rebase"; so started over using https://github.com/dotnet/roslyn-sdk/pull/692

**Pull request closed without merging**